src/classes/ExpectSObject.cls

```diff
@@ -0,0 +1,84 @@
+public class ExpectSObject extends Expect {
+    private static Boolean isSuccess = true;
+    private static Boolean isFail = false;
+    private static String exceptionString;
+    private final SObject expected;
+
+    public ExpectSObject(SObject expected) {
+        this.expected = expected;
+    }
+
+    protected override String getExpectedAsString() {
+        return String.valueOf(expected);
+    }
+
+    public ExpectSObject andIt { get { return this; } }
+
+    public ExpectSObject shouldEqual(SObject actual) {
+        shouldEqual(actual, expectedMessage(actual,''));
+        return this;
+    }
+
+    public ExpectSObject shouldEqual(SObject actual, String message) {
+        assert(expected === actual, message);
+        return this;
+    }
+
+    public ExpectSObject shouldNotEqual(SObject actual) {
+        return shouldNotEqual(actual, expectedMessage(actual,' not'));
+    }
+
+    public ExpectSObject shouldNotEqual(SObject actual, String message) {
+        assert(expected !== actual, message);
```

Comment:
okay awesome. I am fixing the templates to use the constants from the base and to allow inheriting for the string case 😸